packages/steiger-plugin-fsd/src/no-segments-on-sliced-layers/index.ts

```diff
@@ -0,0 +1,34 @@
+import { sep } from 'node:path'
+
+import { getLayers, isSliced, conventionalSegmentNames } from '@feature-sliced/filesystem'
+import type { Diagnostic, Rule } from '@steiger/types'
+
+const noSegmentsOnSlicedLayers = {
+  name: 'no-segments-on-sliced-layers',
+  check(root) {
+    const diagnostics: Array<Diagnostic> = []
+    const layers = Object.values(getLayers(root))
+
+    for (const layer of layers) {
+      if (isSliced(layer)) {
+        for (const directChild of layer.children) {
+          if (directChild.type === 'folder') {
+            const folderName = directChild.path.split(sep).pop()
+            const isConventionalSegment = folderName && conventionalSegmentNames.includes(folderName)
+
+            if (isConventionalSegment) {
+              diagnostics.push({
+                message: `Conventional segment "${folderName}" found as a direct child of a sliced layer. Consider moving it inside a slice or to another layer.`,
```

Comment:
issue: I'm not sure what is meant by "consider moving it to another layer"

suggestion: as I described in a previous comment, there can be two cases — when the intention was for that folder to be a slice, or when the intention was for it to be a segment. If it was supposed to be a slice, a good suggestion, in my opinion, would be to rename it. If it was meant to be a segment, then the suggestion should be to move it inside a slice.
```suggestion
                message: `Conventional segment "${folderName}" should not be a direct child of a sliced layer. Consider moving it inside a slice or, if that is a slice, consider a different name for it to avoid confusion with segments.`,
```

Reply:
Yes, my "consider moving it to another layer" sounds a bit misleading because what I meant was more "consider moving it to the shared layer" for the case where the user puts some highly reusable code that is not related to any slice